src/main/java/org/fungover/haze/Main.java

```diff
@@ -74,14 +74,25 @@ private static void printThreadDebug() {
     }
 
     public static String executeCommand(HazeDatabase hazeDatabase, List<String> inputList) {
+        if (inputList == null || inputList.isEmpty() || inputList.get(0) == null || inputList.get(0).isEmpty()) {
+            return "-ERR no command provided\r\n";
+        }
+
         Log4j2.debug("executeCommand: " + hazeDatabase + " " + inputList);
         String command = inputList.get(0).toUpperCase();
 
-        return switch (command) {
-            case "PING" -> hazeDatabase.ping(inputList);
-            case "SETNX" -> hazeDatabase.setNX(inputList);
-            case "SAVE" -> SaveFile.writeOnFile(hazeDatabase.copy());
-            case "DEL" -> hazeDatabase.delete(inputList.subList(1, inputList.size()));
+        Command commandEnum;
+
+        try {
+            commandEnum = Command.valueOf(command);
+        } catch (IllegalArgumentException ex) {
+            return "-ERR unknown command\r\n";
+        }
+        return switch (commandEnum) {
+            case PING -> hazeDatabase.ping(inputList);
+            case SETNX -> hazeDatabase.setNX(inputList);
+            case SAVE -> SaveFile.writeOnFile(hazeDatabase.copy());
+            case DEL -> hazeDatabase.delete(inputList.subList(1, inputList.size()));
             default -> "-ERR unknown command\r\n";
```

Comment:
```suggestion
```

The default is no longer necessary due to the try/catch :slightly_smiling_face:

Reply:
In this case, I need to have a default case because I cannot use an enum in the switch. Another alternative to use without a default is to write a break after each enum case, but this would make the program break/ we need program fluid it.